materials/material.py

```diff
@@ -0,0 +1,57 @@
+"""Representation of a material's engineering properties and other data."""
+import yaml
+from materials.property import Property, StateDependentProperty
+
+class Material:
+    """An engineering material, in a particular form and condition."""
+    def __init__(self, name, form=None, condition=None, category=None, subcategory=None,
+                 references=None, properties_dict=None):
+        self.name = name
+        self.form = form
+        self.condition = condition
+        self.category = category
+        self.subcategory = subcategory
+        self.references = references
+
+        if properties_dict is not None:
+            self.build_properties(properties_dict)
+
+    def build_properties(self, properties_dict):
+        """Create Property attributes from a (YAML-derived) dictionary."""
+        # Style question: is it bad to dynamically add attributes?
+        # Should I have a dict of Property instead?
+        for property_name, property_dict in properties_dict.items():
+            if 'variation_with_state' in property_dict:
+                prop = StateDependentProperty(property_name, property_dict)
+            else:
+                prop = Property(property_name, property_dict)
+            # TODO check that the property was properly constructed.
+            setattr(self, property_name, prop)
```

Comment:
[factoid] Sometimes it's easier to store and operate on a dict, instead of modifying a class' attributes. Then you can easily iterate over it, determine what is/isn't present, and avoid conflicts with a material property called, for example, `__le__` or something weird like that.

Reply:
changed to dict in https://github.com/mvernacc/material-properties-interchange/pull/1/commits/11a5f967c9f4202695d869ae85e6467e37e0a353